catroid/src/main/java/org/catrobat/catroid/ui/fragment/BrickCategoryFragment.kt

```diff
@@ -0,0 +1,211 @@
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.ui.fragment
+
+import android.content.Context
+import android.os.Bundle
+import android.preference.PreferenceManager
+import android.view.LayoutInflater
+import android.view.Menu
+import android.view.View
+import android.view.ViewGroup
+import android.widget.AdapterView
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.ListFragment
+import org.catrobat.catroid.ProjectManager
+import org.catrobat.catroid.R
+import org.catrobat.catroid.ui.BottomBar.hideBottomBar
+import org.catrobat.catroid.ui.BottomBar.showBottomBar
+import org.catrobat.catroid.ui.BottomBar.showPlayButton
+import org.catrobat.catroid.ui.SpriteActivity
+import org.catrobat.catroid.ui.ViewSwitchLock
+import org.catrobat.catroid.ui.adapter.BrickCategoryAdapter
+import org.catrobat.catroid.ui.addTabLayout
+import org.catrobat.catroid.ui.removeTabLayout
+import org.catrobat.catroid.ui.settingsfragments.AccessibilityProfile
+import org.catrobat.catroid.ui.settingsfragments.RaspberryPiSettingsFragment
+import org.catrobat.catroid.ui.settingsfragments.SettingsFragment
+import org.catrobat.catroid.utils.SnackbarUtil
+import java.util.ArrayList
+import java.util.concurrent.locks.Lock
+
+class BrickCategoryFragment : ListFragment() {
+    private var previousActionBarTitle: CharSequence? = null
+    private var scriptFragment: OnCategorySelectedListener? = null
+    private var adapter: BrickCategoryAdapter? = null
+    private val viewSwitchLock: Lock = ViewSwitchLock()
+
+    companion object {
+        const val BRICK_CATEGORY_FRAGMENT_TAG = "brick_category_fragment"
+    }
+
+    fun setOnCategorySelectedListener(listener: OnCategorySelectedListener) {
+        scriptFragment = listener
+    }
+
+    private fun onlyBeginnerBricks(): Boolean {
+        return PreferenceManager.getDefaultSharedPreferences(activity)
+            .getBoolean(AccessibilityProfile.BEGINNER_BRICKS, false)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val isRestoringPreviouslyDestroyedActivity = savedInstanceState != null
+        if (isRestoringPreviouslyDestroyedActivity) {
+            parentFragmentManager.popBackStack(BRICK_CATEGORY_FRAGMENT_TAG, FragmentManager.POP_BACK_STACK_INCLUSIVE)
+            return
+        }
+        setHasOptionsMenu(true)
```

Comment:
```suggestion
        setHasOptionsMenu(false)
```
Please check if this works, the other solution is a bit odd.